### PR TITLE
Allow array index on graph widget

### LINKED ISSUE #1395: LINEGRAPH widget in TlmViewer Should support array indexing

### PR DESCRIPTION
**Pull request by jmthomas**
closes #1395

### CODE REVIEW

File: openc3-cosmos-init/plugins/packages/openc3-cosmos-demo/targets/INST/cmd_tlm/inst_tlm.txt [hunk 1]
```diff
@@ -49,6 +49,7 @@ TELEMETRY <%= target_name %> HEALTH_STATUS BIG_ENDIAN "Health and status from th
     STATE CONNECTED 1 GREEN
     STATE UNAVAILABLE 0 YELLOW
   APPEND_ITEM BLOCKTEST 80 BLOCK "Block data"
+  APPEND_ITEM BRACKET[0] 8 UINT "Regular item with brackets in the name"
```
Comment (jmthomas) [inline openc3-cosmos-init/plugins/packages/openc3-cosmos-demo/targets/INST/cmd_tlm/inst_tlm.txt:52]: I know it's unusual to add items to the demo but it was the only way to really test it

File: openc3-cosmos-init/plugins/packages/openc3-cosmos-demo/targets/INST/screens/array.txt [hunk 2]
```diff
@@ -5,6 +5,14 @@ VERTICAL
...
   LABELVALUE <%= target_name %> HEALTH_STATUS ARY[0] RAW
   LABELVALUE <%= target_name %> HEALTH_STATUS ARY[1] WITH_UNITS
+  # This item is actually called BRACKET[0] as the name
+  # thus we have to escape the value here to avoid it
+  # being treated like an array item
+  LABELVALUE <%= target_name %> HEALTH_STATUS BRACKET[[0]]
+  LABELSPARKLINE <%= target_name %> HEALTH_STATUS ARY[1]
+    SETTING HISTORY 60s
+  LABELSPARKLINE <%= target_name %> HEALTH_STATUS BRACKET[[0]]
+    SETTING HISTORY 60s
```
Comment (jmthomas) [inline openc3-cosmos-init/plugins/packages/openc3-cosmos-demo/targets/INST/screens/array.txt:15]: Getting all these case to work correctly was interesting

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/GraphWidget.js [hunk 3]
```diff
@@ -115,11 +115,29 @@ export default {
...
         this.screen.addItem(
-          `${item.targetName}__${item.packetName}__${item.itemName}__${item.valueType}`,
-        ),
-      )
+          `${item.targetName}__${item.packetName}__${itemName}__${item.valueType}`,
+          true,
+        )
+        // Return the mapped values since we may have removed bracket escaping
+        return item
+      })
```
Comment (jmthomas) [inline openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/GraphWidget.js:140]: Key to getting this work was reassigning this.items to the mapped version which returns the escaped bracket names

Comment (ryanmelt) [reply openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/GraphWidget.js:140]: This doesn't support an array item with a real bracket in the name.  That's probably ok.

**Review by ryanmelt — APPROVED** [review]

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/GraphWidget.js (hunk 3 above)
Comment (jmthomas) [reply openc3-cosmos-init/plugins/packages/openc3-tool-common/src/components/widgets/GraphWidget.js:140]: If you're naming an array item with brackets then you're really asking for trouble!

**Pull request merged**